File: src/schema/application/temp_connections.ts
import {
  pgTable,
  text,
  integer,
  primaryKey,
  timestamp,
  AnyPgColumn,
  inet,
  // unique,
} from 'drizzle-orm/pg-core';
import {
  system_fields,
  getConfigDefaults,
} from '@dna-platform/crdt-lww-postgres/build/schema/system';
import { table as devices } from './devices';

const config = (table) => ({
  pk: primaryKey({ columns: [table.id, table.timestamp] }),
  ...getConfigDefaults.defaultIndexes('temp_connections', table),
});

export const table = pgTable(
  'temp_connections',
  {
    ...system_fields,
    id: text('id'), // Primary key
    timestamp: timestamp('timestamp', { withTimezone: true }), // Connection Time
    hypertable_timestamp: text('hypertable_timestamp'), // Hypertable timestamp
    interface_name: text('interface_name'), // Network Interface Name

    total_packet: integer('total_packet'), // Total packets in the connection
    total_byte: integer('total_byte'), // Total bytes in the connection
    device_id: text('device_id').references(() => devices.id as AnyPgColumn), // Device ID

    protocol: text('protocol'), // Protocol (TCP, UDP, ICMP, etc.)
    source_ip: inet('source_ip'), // Source IP Address
    destination_ip: inet('destination_ip'), // Destination IP Address

    source_port: integer('source_port'), // Source Port (Optional)
    destination_port: integer('destination_port'), // Destination Port (Optional)
    remote_ip: text('remote_ip'), // Remote IP Address
  },
  config,
);

File: bin/store/src/database/schema/tables/temp_connections.rs
use crate::define_table_schema;
use crate::generated::schema::generator::diesel_schema_definition::{
    types::*, DieselTableDefinition,
};
use crate::{system_fields, system_foreign_keys, system_indexes};

/**
 * Auto-converted from TypeScript schema source.
 */
pub struct TempConnectionsTable;

define_table_schema! {
    hypertable: false,
    fields: {
        // System fields - common across all tables ( REQUIRED )
        system_fields!(),

        hypertable_timestamp: nullable(text()),
        interface_name: nullable(text()),
        total_packet: nullable(integer()),
        total_byte: nullable(integer()),
        device_id: nullable(text()),
        protocol: nullable(text()),
        source_ip: nullable(text()),
        destination_ip: nullable(text()),
        source_port: nullable(integer()),
        destination_port: nullable(integer()),
        remote_ip: nullable(text()),
    },
    indexes: {
        // System field indexes ( REQUIRED )
        system_indexes!("temp_connections"),
    },
    foreign_keys: {
        // System field foreign keys ( REQUIRED )
        system_foreign_keys!("temp_connections"),
        fk_temp_connections_device_id: { columns: ["device_id"], foreign_table: "devices", foreign_columns: ["id"], on_delete: "no action", on_update: "no action" },
    }
}
